File: loc_phat_3.cpp
#include<iostream>
#include<string>

using namespace std;

int main(){
    int t;
    cin>>t;
    while(t--){
        int n;
        cin>>n;
        int sum=0, mul=2;
        for(int i=1;i<=n;i++){
            sum+=mul;
            mul*=2;
        }
        cout<<sum<<endl;
        for(int i=n;i>0;i--){
            string res="", last="";
            for(int j=0;j<i;j++){
                res+='8';
                last+='6';
            }
            cout<<res<<' ';
            while(res!=last){
                for(int j=i-1;j>=0;j--){
                    if(res[j]=='8'){                              
                        res[j]='6';                                 
                        break;
                    }
                    if(res[j]='6'){
                        res[j]='8';
                        continue;
                    }
                }
                cout<<res<<' ';
            }
        }
        cout<<endl;
    }
    return 0;
}

File: loc_phat_3.py
t=int(input())

for v in range(t):
    n=int(input())
    sum, mul=0, 2
    for h in range(n):
        sum+=mul
        mul*=2
    print(sum)
    for i in range(n,0,-1):
        res, last='',''
        for j in range(i):
            res+='8'
            last+='6'
        print(res,end=' ')
        while res!=last:
            for j in range(i-1,-1,-1):
                if(res[j]=='8'):
                    tmp=list(res)
                    tmp[j]='6'
                    res=''.join(tmp)
                    break
                if(res[j]=='6'):
                    tmp=list(res)
                    tmp[j]='8'
                    res=''.join(tmp)
                    continue
            print(res,end=' ')
    print()